lib/iris/_lazy_data.py

```diff
@@ -47,7 +47,14 @@ def is_lazy_data(data):
     return result
 
 
-def _optimum_chunksize(chunks, shape, limit=None, dtype=np.dtype("f4")):
+@lru_cache
+def _optimum_chunksize_internals(
+    chunks,
+    shape,
+    limit=None,
+    dtype=np.dtype("f4"),
+    dask_array_chunksize=dask.config.get("array.chunk-size"),
```

Comment:
The fetching of a default dask chunksize here really **_needs_** to be dynamic.
It's no good defaulting to "some value copied from the dask setting at some time", since the whole point of using it is that we might want to set it to different values to control the loading behaviour.

That could make this whole approach invalid for this routine??
I guess you need another layer to translate 'limit == None' as "use the dask chunksize",  which then takes the (current) dask-chunksize as a fixed value -- which you can then memoize.